### PR TITLE
feat/check auth chat & implement join chat room

### PR DESCRIPTION
**Pull request by sieuhuflit**
## Overview

- Implement auth check when try access chat page
- Fix UI for display wrong online badge 
- Add future_loading_dialog based from this package : [Link](https://pub.dev/packages/future_loading_dialog)
- Implement join room when got invite

## Demo

Sep-13-2023 12-56-36

### CODE REVIEW

File: lib/core/presentation/pages/home/home_page.dart [hunk 1]
```diff
@@ -50,7 +50,12 @@ class _HomePageState extends State<HomePage> {
           GestureDetector(
             behavior: HitTestBehavior.translucent,
             onTap: () {
-              AutoRouter.of(context).navigate(const ChatListRoute());
+              final authState = BlocProvider.of<AuthBloc>(context).state;
+              if (authState is AuthStateAuthenticated) {
```
Comment (mthinh) [inline lib/core/presentation/pages/home/home_page.dart:54]: we can use state.maybeWhen(authenticated: () => doSth, orElse: showLogin)

**Review by mthinh — CHANGES_REQUESTED** [review]
@sieuhuflit pls help check

File: lib/core/presentation/pages/home/home_page.dart (hunk 1 above)
Comment (sieuhuflit) [reply lib/core/presentation/pages/home/home_page.dart:54]: Thanks 👍

**Comment by mthinh** [comment]
LGTM 👍

**Review by mthinh — APPROVED** [review]

**Pull request merged**